arangod/RocksDBEngine/RocksDBRestExportHandler.cpp

```diff
@@ -103,72 +102,103 @@ VPackBuilder RocksDBRestExportHandler::buildOptions(VPackSlice const& slice) {
   } else {
     options.add("batchSize", VPackValue(1000));
   }
-
-  VPackSlice limit = slice.get("limit");
-  if (limit.isNumber()) {
-    options.add("limit", limit);
-  }
-
-  VPackSlice flush = slice.get("flush");
-  if (flush.isBool()) {
-    options.add("flush", flush);
-  } else {
-    options.add("flush", VPackValue(false));
-  }
-
+  
   VPackSlice ttl = slice.get("ttl");
   if (ttl.isNumber()) {
     options.add("ttl", ttl);
   } else {
     options.add("ttl", VPackValue(30));
   }
-
-  VPackSlice flushWait = slice.get("flushWait");
-  if (flushWait.isNumber()) {
-    options.add("flushWait", flushWait);
-  } else {
-    options.add("flushWait", VPackValue(10));
+  
+  uint64_t limit = UINT64_MAX;
+  VPackSlice limitSlice = slice.get("limit");
+  if (limitSlice.isNumber()) {
+    limit = limitSlice.getInt();
   }
-  options.close();
-
+  
+  options.add("options", VPackValue(VPackValueType::Object));
+  options.add("stream", VPackValue(true)); // important!!
+  VPackSlice count = slice.get("count");
+  if (count.isBool() && count.getBool()) {
+    
+    auto ctx = transaction::StandaloneContext::Create(_vocbase);
+    SingleCollectionTransaction trx(ctx, cname, AccessMode::Type::READ);
+    Result res = trx.begin();
+    
+    if (res.fail()) {
+      THROW_ARANGO_EXCEPTION(res);
+    }
+    OperationResult opRes = trx.count(cname, true);
+    trx.finish(opRes.result);
+    // QueryStreamCursor will add `exportCount` as `count`
+    options.add("exportCount", VPackValue(std::min(limit, opRes.slice().getUInt())));
```

Comment:
what exactly is this for?
if the number of documents is queried here, it may be changed later on before the actual export starts, so there may be data races.